packages/wonder-blocks-data/util/request-tracking.js

```diff
@@ -57,25 +87,61 @@ export class RequestTracker {
         if (this._trackedRequests[type][key] == null) {
             this._trackedRequests[type][key] = options;
         }
-    }
+    };
+
+    /**
+     * Reset our tracking info.
+     */
+    reset = () => {
+        this._trackedHandlers = {};
+        this._trackedRequests = {};
+    };
 
     /**
-     * This method is temporary. Will be replaced with the fulfilAll... call when
-     * that is implemented.
+     * Initiate fulfillment of all tracked requests.
      *
-     * TODO(jeff): Delete this when the fulfilment method is implemented.
+     * This loops over the requests that were tracked using TrackData, and asks
+     * the respective handlers to fulfill those requests in the order they were
+     * tracked.
+     *
+     * Calling this method marks tracked requests as fulfilled; requests are
+     * removed from the list of tracked requests by calling this method.
+     *
+     * @returns {Promise<Cache>} A frozen cache of the data that was cached
+     * as a result of fulfilling the tracked requests.
      */
-    tempGetTrackedRequestsAndHandlers() {
-        return {
-            trackedHandlers: this._trackedHandlers,
-            trackedRequests: this._trackedRequests,
-        };
-    }
-}
+    fulfillTrackedRequests = (): Promise<$ReadOnly<Cache>> => {
+        const promises = [];
 
-/**
- * The default export is the instance of the request tracker that each of our
- * other pieces use. This way everything uses the same interface but we have the
- * class with which to perform testing.
- */
-export default new RequestTracker();
+        for (const handlerType of Object.keys(this._trackedHandlers)) {
+            const handler = this._trackedHandlers[handlerType];
+
+            // For each handler, we will perform the request fulfillments!
+            const requests = this._trackedRequests[handlerType];
+            for (const requestKey of Object.keys(requests)) {
+                const promise = this._requestFulfillment.fulfill(
+                    handler,
+                    requests[requestKey],
+                );
+
+                /**
+                 * We shouldn't get bad results here, but we might, so let's
+                 * be defensive.
+                 */
+                if (promise != null) {
+                    promises.push(promise);
+                }
+            }
+        }
+
+        /**
+         * Clear out our tracked info.
+         */
+        this.reset();
+
+        /**
+         * Let's wait for everything to fulfill, and then clone the cached data.
+         */
+        return Promise.all(promises).then(() => this._responseCache.clone());
```

Comment:
To make sure that we don't end up with some new requests that aren't in the promise set but have since been tracked. I want the API to be easy to reason about and resetting after the promises resolve, or not resetting at all is far more confusing an API to think about.

Reply:
I've added a comment to clarify.

Reply:
That's a pretty good reason. 🙂Thanks for adding a comment.